### DISCUSSION TITLE
Support for midi loopback devices [renamed]

### DISCUSSION THREAD

**Discussion by aeantipov** [helgoboss/helgobox#1188] [Q&A] [answered]
Hi @helgoboss, 

Happy customer here. A question. I was setting up my reaper project to route audio/midi through virtual devices (Reaper aux loopback and Reaper midi loopback), so that I can control the inputs on my midi controller, making an input mixer. However, I noticed Playtime doesn't allow input from aux loopback devices (it does for midi) which breaks my flow. Is that something you plan to add anytime soon?

**Comment by helgoboss** [comment, marked as answer]
It's fixed in 2.16.4.

**Reply by aeantipov** [reply]
Thanks. Related (don't know if you want a separate thread) - what about receives? Afaik now you're required to give a hardware input.

**Reply by helgoboss** [reply]
Recording audio from receives will probably come along with #999. Recording MIDI from receives ... I'd need a strong reason to implement that. Haven't found any compelling use case so far (features of this kind add lots of hidden complexity to Playtime, so I will only consider it if there's some good reason for it). If you need that, please create a new discussion in the "Ideas" section along with some background why you think that's worth it.

**Reply by aeantipov** [reply]
Well, probably if you have loopback devices that you don't need sends - anything can be routed through these virtual devices.

**Comment by aeantipov** [comment]
Hm, actually this still doesn't work for me
![image](https://github.com/user-attachments/assets/2dd546a6-f00e-4769-b885-febe9b09138d)
Reaper 7.22 (not sure how to get 7.23+), Playtime 2.16.4 installed via Reapack, 
Loopback devices enabled via 
```
[reaper]
rearoute_loopback=8
```
in reaper.ini

![image](https://github.com/user-attachments/assets/67ecb84f-0734-4d55-a0f0-83b8724f27f0)

**Reply by helgoboss** [reply]
What, you mean audio loopbacks? What is that? I only know MIDI loopbacks.

**Reply by aeantipov** [reply]
Reaper actually now has native audio loopbacks which, I realized, work in the same project tab and they are fast. 
https://reaper.blog/2022/10/669-update/

**Reply by aeantipov** [reply]
This in principle would allow you to bypass the whole send/receive audio thing.

**Reply by helgoboss** [reply]
Oh ok. New to me as well. If you want support for that, please open a feature request (idea category). But I'm not sure now if it's even possible to add support for that, depends on REAPER's API support.

Definitely not going to happen soon, my priority is the final release.

**Reply by helgoboss** [reply]
What's the difference to sends/receices then? Advantages/disadvantages? Isn't there a catch concerning latency and stuff?

**Reply by aeantipov** [reply]
The big thing is you can arm tracks for record, whereas for send receives they will always pass audio unless you mute them. Latency - I would need to measure, but I think both are fast.

**Reply by aeantipov** [reply]
Done - see #1190
